go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -280,6 +280,197 @@ func (c *Controller) reconcile(name string) error {
 	return c.updateRAStatus(ra, hadUpdates, err)
 }
 
+func subnetsOverlap(subnet1, subnet2 string) (bool, error) {
+	ipNets, err := util.ParseIPNets([]string{subnet1, subnet2})
+	if err != nil {
+		return false, err
+	}
+
+	// Two subnets overlap if one contains the other's IP or vice versa
+	return ipNets[0].Contains(ipNets[1].IP) || ipNets[1].Contains(ipNets[0].IP), nil
+}
+
+// checkSubnetOverlaps validates that subnets from the current RA don't overlap
+// with any existing subnets in the same VRFs, both within the RA itself and
+// across different RAs
+func (c *Controller) checkSubnetOverlaps(ra *ratypes.RouteAdvertisements, selectedNetworks *selectedNetworks) error {
+	NetworkVRFsSubnets := createNetworkVRFsSubnetsMap(ra, selectedNetworks)
+
+	for networkVRF, currentRASubnets := range NetworkVRFsSubnets {
+		// Get all other RAs and their subnets for the same VRF
+		allRAs, err := c.raLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+
+		foundOtherRA := false
+		for _, otherRA := range allRAs {
+			if otherRA.Name == ra.Name {
+				continue
+			}
+			otherSelectedNetworks, err := c.getSelectedNetworksForRA(otherRA)
+			if err != nil {
+				// If we can't get networks from another RA, log but don't fail current RA reconcile.
+				// This could happen if the other RA has configuration issues
+				klog.V(4).Infof("Failed to get selected networks from RouteAdvertisements %q: %v", otherRA.Name, err)
+				continue
+			}
+
+			otherRANetworkVRFsSubnets := createNetworkVRFsSubnetsMap(otherRA, otherSelectedNetworks)
+			otherSubnets, hasMatchingVRF := otherRANetworkVRFsSubnets[networkVRF]
+			if !hasMatchingVRF || len(otherSubnets) == 0 {
+				continue
+			}
+
+			foundOtherRA = true
+			combinedSubnets := append(currentRASubnets, otherSubnets...)
+			overlappingIndices, err := checkSubnetSliceOverlapsWithIndices(combinedSubnets)
+			if err != nil {
+				return fmt.Errorf("%w: failed to check subnet overlap: %w", errConfig, err)
+			}
+
```

Comment:
Let's remove identation

```golang
if len(overlappingIndices) == 0 {
   continue
}
```

Reply:
not sure I follow

Reply:
if there is no overlapping we continue with the loop, this way we don't need to indent under if len(... all the checking code.